certification/engine/operatorsdk.go

```diff
@@ -0,0 +1,188 @@
+package engine
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/errors"
+	fileutils "github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/utils/file"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
+	log "github.com/sirupsen/logrus"
+)
+
+func NewOperatorSdkEngine() *cli.OperatorSdkEngine {
+	var engine cli.OperatorSdkEngine = operatorSdkEngine{}
+	return &engine
+}
+
+type operatorSdkEngine struct{}
+
+func (o operatorSdkEngine) Scorecard(image string, opts cli.OperatorSdkScorecardOptions) (*cli.OperatorSdkScorecardReport, error) {
+	cmdArgs := []string{"scorecard"}
+	if opts.OutputFormat == "" {
+		opts.OutputFormat = "json"
+	}
+	cmdArgs = append(cmdArgs, "--output", opts.OutputFormat)
+	if opts.Selector != nil {
+		for _, selector := range opts.Selector {
+			cmdArgs = append(cmdArgs, fmt.Sprintf("--selector=%s", selector))
+		}
+	}
+	if opts.Kubeconfig != "" {
+		cmdArgs = append(cmdArgs, "--kubeconfig", opts.Kubeconfig)
+	}
+	if opts.Namespace != "" {
+		cmdArgs = append(cmdArgs, "--namespace", opts.Namespace)
+	}
+	if opts.ServiceAccount != "" {
+		cmdArgs = append(cmdArgs, "--service-account", opts.ServiceAccount)
+	}
+
+	configFile, err := createScorecardConfigFile()
+	if configFile != "" {
+		defer os.Remove(configFile)
+	}
```

Comment:
Just did a quick test and it seems os.Remove returns an error that the directory does not exist. I think you can remove the if block and ignore the returned error.

Reply:
This will be handled in #188